Fix thumbnail and image to be separated

tomaz82 opened the pull request:
Fix thumbnail and image to be separated as I think it was originally meant to be.

The current implementation for VideoViewis that md_video can show a thumbnail before fading to the video.
This is specified with <thumbnail> tag in the metadata, or if that doesn't contain a valid image, it tries with <image>.

Ontop of this a theme can still have the md_image element, which is where I believe the current implementation is broken, because that one will also use thumbnail, where I believe it should only use what's specified in <image>. So that the video thumbnail in md_video, and the actual md_image element, can be 2 different textures. I'm thinking screenshot maybe for the thumbnail, and boxart for the image. Current code would use the screenshot on both elements.

Anyway, long description. This PR fixes that.

EctoOne commented:
It seems the description is missing two things for me, so I have to ask.
Does this mean that we can add the <thumbnail> tag in the gamelist.xml and it will show that image and not the one from the <image> tag if video is delayed?
Also, because that sentence is cut off for me, what image is shown when we don't have the <thumbnail> tag?
And what happens if we don't have a <video> specified but a <thumbnail>, would the image from the <thumbnail> still show? I personally use the <marquee> tag to display screenshots right now (because I don't want and can't use videos for different reasons). So with this fix I'm guessing that I can just add the <thumbnail> tag to my gamelists, and therefore can modify my theme to show boxart (md_image), logo (<md_marquee>) and screenshot (md_video) when I have no <video> tag for my entries and force Emulationstation to use video view.

Darknior commented:
I'm really interested to understand this too please.
Nobody explain THUNMNAIL in ES Theme doc : https://github.com/RetroPie/EmulationStation/blob/master/THEMES.md

How can we use it : 
- Is it an normal image element TYPE / MEDIA ?
- What VIEW can use it ? Detail + Video ? Grid ?
- How can we call it ? `<image name="md_thumbnail">`
- Is it used to replace video if no video with `<showSnapshotNoVideo>true</showSnapshotNoVideo>`

Because it can be used to replace video by screenshot, and let us show at same time the Cover and Logo, great update :)
And it can be also use on GRID view to make it loading faster with mini-screenshot.

Thanks

tomaz82 commented:
I fixed my description to use the code tags so `<>` should show properly now.

Okay, so it's like this: md_video can have an image show if video is missing or delayed. That has always been using `<thumbnail>` if that existed in the gamelist.xml or `<image>` if `<thumbnail>` was missing. The problem here is that md_image was always using the same image as md_video, no matter if it was the `<image>` or `<thumbnail>`. This is now fixed so md_image always use whatever is in `<image>`.

So an example, if the gamelist.xml contains cover art in `<image>` and a screenshot in `<thumbnail>` then the md_video with delayed video ( or missing `<video>` ) will show the screenshot, and md_image will show the cover art. If the `<thumbnail>` is missing then md_video will show the cover art.

Makes more sense now?

Darknior commented:
Ok thanks ... it make sense for me now :)

Is it possible to also add `<image name="md_thumbnail">` to let us use the screenshot where we want ?
Not only to replace the video if missing ... with same VIEW & TYPE than `md_image`
It can be cool to have one more media to use :)
Thanks

tomaz82 commented:
There is no md_thumbnail no and it will most likely not be added since it's meant to be a thumbnail for whenever the video is missing. There are a couple other new features planned tho that MIGHT be coming soon(tm)

Darknior commented:
I have try it and like i can't declare the element to give it a POS ORIG SIZE etc .... i have a bug :(
If you don't want to add it, can you add a quick fix to make it **using the VIDEO attributes** ?
Thanks

Before
![20180201_124859_hdr](https://user-images.githubusercontent.com/3703317/35677175-df09a848-074e-11e8-9f9d-15cbdd8d9595.jpg)
Screenshot with **Thumnail**
![20180201_124903_hdr](https://user-images.githubusercontent.com/3703317/35677188-ebe13752-074e-11e8-9745-b51465271394.jpg)
**Video**
![20180201_124841_hdr](https://user-images.githubusercontent.com/3703317/35677189-ef2d6a3e-074e-11e8-878f-e0ac755b597a.jpg)

Darknior commented:
You already use ORIG and POSITION you only forgot SIZE ;)

tomaz82 commented:
Sorry, I'm not sure what you are saying.

Darknior commented:
On my Theme i write : 

```
			<video name="md_video">
				<origin>0 0</origin>
				<pos>0.3833 0.242</pos>
				<size>0.3344 0.4452</size>
				
				<!-- <default> PATH </default> -->
				<showSnapshotNoVideo>true</showSnapshotNoVideo> <!-- Show thumbnail from gamelist.xml if no video and no default -->
				<showSnapshotDelay>true</showSnapshotDelay>
				<delay>0.5</delay> <!-- Second -->
				<zIndex>50</zIndex> <!-- 30 by Default forced to 5000 by OMX -->
			</video>
```
To show the THUMBNAIL you are using the good VIDEO **ORIGIN** and **POS**.
But you don't use the **SIZE** and the thumbnail use a size i don't choose, i don't understand where he take it, i think it is a default size.
Like you can see on my second screenshot the thumbnail is really to big ...
- Can you force it the **VIDEO SIZE**
- Or add a new attribute for it `<snapsize>` ... but i think it's less good.

EctoOne commented:
Instead of using just `<size>`, I'm using:
```  <video name="md_video">
   <maxSize>0.2 0.2</maxSize>
  </video>
```
And it works fine.

Darknior commented:
Thanks @EctoOne . I don't try it before and it works fine :)
But if possible it can be cool to also use the SIZE attribute if there is no MAXSIZE.
For my part i want to force all my videos to have the same size, maxSize don't do that.

tomaz82 commented:
I'll look into it

tomaz82 commented:
Ran several tests and I find no problems, md_video with a size tag works just fine for the thumbnail.
